includes/discovery/functions.inc.php

```diff
@@ -1097,8 +1097,15 @@ function discovery_process(&$valid, $device, $sensor_type, $pre_cache)
             foreach ($raw_data as $index => $snmp_data) {
                 // get the value for this sensor, check 'value' and 'oid', if state string, translate to a number
                 $data_name = isset($data['value']) ? $data['value'] : $data['oid'];  // fallback to oid if value is not set
-                if (is_numeric($snmp_data[$data_name])) {
-                    $value = $snmp_data[$data_name];
+
+                $tmp_value = $snmp_data[$data_name];
+                preg_match('/[\d\.\-]+/', $tmp_value, $temp_response);
```

Comment:
I'm guessing something returns a string containing a value and text?

I'd prefer the is_numeric() test first so we can avoid the preg_match for most things.

Reply:
Also, perhaps this would be a better regex: `-?\d*\.?\d+`

It only allows a single - at the start and a single decimal point, possibly at the start

Reply:
Thanks, pushing an update now.